tests/test_perf_analyzer_config_generator.py

```diff
@@ -0,0 +1,284 @@
+            yaml_content, expected_configs, '-c 1,2,3,4')
+
+    def test_batch_size_and_concurrency(self):
+        '''
+        Test:
+            - Schmoo batch sizes and concurrency
+
+        Batch sizes: 1,2,4
+        Concurrency: 1-4
+
+        12 configs will be generated
+        '''
+
+        # yapf: disable
+        yaml_content = convert_to_bytes("""
+            profile_models:
+                - my-model
+            """)
+        # yapf: enable
+
+        batch_sizes = [1, 2, 4]
+        concurrencies = [1, 2, 3, 4]
+
+        expected_configs = [
+            self._create_expected_config(batch_size=b, concurrency=c)
+            for b in batch_sizes
+            for c in concurrencies
+        ]
+
+        pa_cli_args = ['-b 1,2,4', '-c 1,2,3,4']
+        self._run_and_test_perf_analyzer_config_generator(
+            yaml_content, expected_configs, pa_cli_args)
+
+    def _create_expected_config(self,
+                                batch_size=DEFAULT_BATCH_SIZES,
+                                concurrency=DEFAULT_CONCURRENCY,
+                                launch_mode=DEFAULT_TRITON_LAUNCH_MODE,
+                                client_protocol=DEFAULT_CLIENT_PROTOCOL):
+        expected_config = PerfAnalyzerConfig()
+        expected_config._options['-m'] = 'my-model'
+        expected_config._options['-b'] = batch_size
+        expected_config._args['concurrency-range'] = concurrency
+        expected_config._args['measurement-mode'] = DEFAULT_MEASUREMENT_MODE
+
+        if launch_mode == 'c_api':
+            expected_config._args['service-kind'] = 'triton_c_api'
+            expected_config._args[
+                'triton-server-directory'] = DEFAULT_TRITON_INSTALL_PATH
+            expected_config._args[
+                'model-repository'] = DEFAULT_OUTPUT_MODEL_REPOSITORY
+        else:
+            expected_config._options['-i'] = client_protocol
+            if client_protocol == 'http':
+                expected_config._options['-u'] = DEFAULT_TRITON_HTTP_ENDPOINT
+            else:
+                expected_config._options['-u'] = DEFAULT_TRITON_GRPC_ENDPOINT
+
+        return expected_config
+
+    def _run_and_test_perf_analyzer_config_generator(self,
+                                                     yaml_content,
+                                                     expected_configs,
+                                                     pa_cli_args=None):
+        args = [
+            'model-analyzer', 'profile', '--model-repository', 'cli_repository',
+            '-f', 'path-to-config-file'
+        ]
+
+        if type(pa_cli_args) == list:
+            args = args + pa_cli_args
+        elif type(pa_cli_args) == str:
+            args.append(pa_cli_args)
+
+        config = self._evaluate_config(args, yaml_content)
+
+        # Assign concurrency to one if not specified on the CLI
+        if not config.concurrency:
+            config.concurrency = [1]
```

Comment:
Is this the real behavior of the system? I don't think it is, so we shouldn't be doing this here in the test.

Reply:
Removed